core/DataTable/Map.php

```diff
@@ -540,4 +540,25 @@ public function getColumns()
         }
         return array();
     }
+
+
+    /**
+     * Return the metadata of the latest item in the datable array.
+     * @return array
+     * @throws Exception
+     */
+    public function getLatestMetaData()
+    {
+
+        $largestKey = null;
+        // find the largest key
+        foreach ($this->array as $key => $item) {
+            if (!$largestKey || (strtotime($largestKey) < strtotime($key))) {
```

Comment:
This won't work in all cases. Datatable maps can be indexed by date, which you seem to assume, but they can also be index by idsite. And actually it can also be a map of maps, on indexed by date, the other by idsite.
As it might be hard to handle all those cases here in a global method that is actually only needed for one special purpose, I would still recommend to move that handling to the Visualization class instead, where you can be sure it can't be a map of maps or indexed by idsite.
Also iterating over the index doesn't actually fix what I mentioned before. An older date can still have a newer generation date, due to invalidations.